src/ringmesh/io/geomodel_builder_resqml.cpp

```diff
@@ -0,0 +1,968 @@
+                    {
+                        read_property( *propVal, patch,
+                            cur_surf.vertex_attribute_manager() );
+                    }
+                    else if( attachment
+                             == gsoap_resqml2_0_1::
+                                    resqml2__IndexableElements__cells )
+                    {
+                        read_property( *propVal, patch,
+                            cur_surf.polygon_attribute_manager() );
+                    }
+                }
+
+                global_point_count += pointCount;
+            }
+        }
+
+        builder_.build_lines_and_corners_from_surfaces();
+        builder_.build_regions_from_lines_and_surfaces();
+        builder_.geology.build_contacts();
+        return true;
+    }
+
+    namespace
+    {
+        bool read_volume_rep( VolumeMesh3D& mesh,
+            UnstructuredGridRepresentation& unstructed_grid )
+        {
+            auto mesh_builder = VolumeMeshBuilder3D::create_builder( mesh );
+
+            unstructed_grid.loadGeometry();
+
+            std::unique_ptr< double[] > gridPoints(
+                new double[unstructed_grid.getXyzPointCountOfPatch( 0 ) * 3] );
+            unstructed_grid.getXyzPointsOfAllPatchesInGlobalCrs(
+                &gridPoints[0] );
+
+            const ULONG64 nb_vertices =
+                unstructed_grid.getXyzPointCountOfPatch( 0 );
+
+            for( auto v : range( nb_vertices ) )
+            {
+                const vec3 vertex( gridPoints[v * 3], gridPoints[v * 3 + 1],
+                    gridPoints[v * 3 + 2] );
+
+                mesh_builder->create_vertex( vertex );
+            }
+
+            const ULONG64 nb_cells = unstructed_grid.getCellCount();
+            for( auto c : range( nb_cells ) )
+            {
+                const index_t nb_faces =
+                    (index_t) unstructed_grid.getFaceCountOfCell( c );
+                if( nb_faces == 4 )
+                {
+                    const index_t cell = mesh_builder->create_cells(
+                        (index_t) 1, CellType::TETRAHEDRON );
+
+                    std::vector< index_t > vertices = {
+                        (index_t) unstructed_grid.getNodeIndicesOfFaceOfCell(
+                            cell, 0 )[0],
+                        (index_t) unstructed_grid.getNodeIndicesOfFaceOfCell(
+                            cell, 0 )[1],
+                        (index_t) unstructed_grid.getNodeIndicesOfFaceOfCell(
+                            cell, 0 )[2],
+                        0
+                    };
+
+                    bool found = false;
+                    for( unsigned int f = 1; f < nb_faces; ++f )
+                    {
+                        const ULONG64 nb_nodes =
+                            unstructed_grid.getNodeCountOfFaceOfCell( cell, f );
+
+                        for( ULONG64 node = 0; node < nb_nodes; ++node )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of nested break and goto statements from 2 to 1 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3ASingleGotoOrBreakPerIteration)